packages/zowe-explorer/src/trees/job/JobFSProvider.ts

```diff
@@ -213,13 +223,23 @@ export class JobFSProvider extends BaseProvider implements vscode.FileSystemProv
 
         const jesApi = ZoweExplorerApiRegister.getJesApi(spoolEntry.metadata.profile);
         await AuthHandler.waitForUnlock(spoolEntry.metadata.profile);
+        const query = new URLSearchParams(uri.query);
+        let recordRange = "";
+        if (query.has("startLine")) {
+            const startLine = parseInt(query.get("startLine")!);
+            const endLine = query.has("endLine")
+                ? parseInt(query.get("endLine")!)
+                : SettingsConfig.getDirectValue<number>("zowe.jobs.recordsToFetch") ?? 0;
+            recordRange = `${startLine}-${endLine}`;
+        }
         try {
```

Comment:
When the setting `zowe.jobs.recordsToFetch` is set, can we only grab the first X lines (configured by this setting)?
Otherwise, the whole job spool is fetched when its opened for the first time.